pkg/utils/selector.go

```diff
@@ -221,6 +245,79 @@ func CheckPodMeetSelector(pod v1.Pod, selector v1alpha1.SelectorSpec) (bool, err
 	return false, nil
 }
 
+// filterByNode filters a list of pods by a given node selector.
+func filterByNodes(pods []v1.Pod, nodes labels.Selector) ([]v1.Pod, error) {
```

Comment:
Can we merge `filterByNodes` and  `filterPodByNodeSelector` together?  First we select node by node selector. Then we discard the pod which is not set on the selected node set. I think this would be quite clear and easy to understand.

Reply:
Done.